app/main.py

```diff
@@ -1,2 +1,64 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(
+            self,
+            capacity: int = 8,
+            load_factor: float = 2 / 3
+    ) -> None:
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.table = [None] * self.capacity
+
+    def index_(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+        while self.table[index] and self.table[index][1] != key:
+            index += 1
+            index %= self.capacity
+        return index
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self.index_(key)
+        if not self.table[index]:
+            raise KeyError(f"Key {key} is not in dictionary")
+        return self.table[index][2]
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size > self.capacity * self.load_factor:
+            self.resize()
+
+        index = self.index_(key)
+        if not self.table[index]:
+            self.size += 1
+        self.table[index] = [hash(key), key, value]
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = self.index_(key)
+        if self.table[index]:
+            self.table[index] = [None]
+            self.size -= 1
+
+    def resize(self) -> None:
+        temp_table = self.table
+        self.capacity *= 2
+        self.table = [None] * self.capacity
+        self.size = 0
+        for element in temp_table:
+            if element:
+                self.__setitem__(element[1], element[2])
```

Comment:
```suggestion
                self[element[1]] = element[2]
```

Reply:
can you comment in more detail?

Reply:
can you comment in more detail?

Reply:
`self[element[1]] = element[2]` is a more straightforward and concise way to set a value for a specific key. It directly accesses the dictionary element and is self-explanatory.

Reply:
but i can set value with setter, is it a mistake to not accept the request?